packages/ui/test/accounts/hooks/useHasRequiredStake.test.tsx

```diff
@@ -0,0 +1,200 @@
+import { ApiContext } from '@/common/providers/api/context'
+
+import { alice, aliceStash, bob } from '../../_mocks/keyring'
+import { MockKeyringProvider } from '../../_mocks/providers'
+import { stubApi } from '../../_mocks/transactions'
+
+const useAccounts: { hasAccounts: boolean; allAccounts: Account[] } = {
+  hasAccounts: true,
+  allAccounts: [],
+}
+
+jest.mock('../../../src/accounts/hooks/useAccounts', () => {
+  return {
+    useAccounts: () => useAccounts,
+  }
+})
+
+let balance: Balances | null = null
+
+const useBalance = {
+  useBalance: () => balance,
+}
+
+jest.mock('../../../src/accounts/hooks/useBalance', () => useBalance)
+
+let balances: AddressToBalanceMap = {}
+
+const useBalances = {
+  useBalances: () => balances,
+}
+
+jest.mock('../../../src/accounts/hooks/useBalances', () => useBalances)
+
+describe('useHasRequiredStake', () => {
+  const useApi = stubApi()
+
+  jest.useFakeTimers()
+
+  beforeAll(async () => {
+    useAccounts.hasAccounts = true
+    useAccounts.allAccounts.push(alice, aliceStash, bob)
+  })
+
+  afterEach(cleanup)
+
+  function renderUseTotalBalances(stake: number) {
+    const wrapper = ({ children }: { children: ReactNode }) => (
+      <MockKeyringProvider>
+        <AccountsContextProvider>
+          <ApiContext.Provider value={useApi}>{children}</ApiContext.Provider>
+        </AccountsContextProvider>
+      </MockKeyringProvider>
+    )
+    return renderHook(() => useHasRequiredStake(stake), { wrapper })
+  }
+
+  it('Member has enough founds', () => {
+    balance = {
+      total: new BN(1000),
+      locked: new BN(0),
+      transferable: new BN(1000),
+      recoverable: new BN(0),
+      locks: [],
+    }
+    balances = {
+      [alice.address]: balance,
+      [aliceStash.address]: {
+        total: new BN(1000),
+        locked: new BN(0),
+        transferable: new BN(1000),
+        recoverable: new BN(0),
+        locks: [],
+      },
+      [bob.address]: {
+        total: new BN(1000),
+        locked: new BN(0),
+        transferable: new BN(1000),
+        recoverable: new BN(0),
+        locks: [],
+      },
+    }
+    const { result } = renderUseTotalBalances(1000)
+    expect(result.current).toStrictEqual({
+      hasRequiredStake: true,
+      transferableAccounts: null,
+      accountsWithLockedFounds: null,
+    })
+  })
+
+  it('Member has not enough founds', () => {
+    balance = {
+      total: new BN(10),
+      locked: new BN(0),
+      transferable: new BN(10),
+      recoverable: new BN(0),
+      locks: [],
+    }
+    balances = {
+      [alice.address]: balance,
+      [aliceStash.address]: {
+        total: new BN(50),
+        locked: new BN(0),
+        transferable: new BN(50),
+        recoverable: new BN(0),
+        locks: [],
+      },
+      [bob.address]: {
+        total: new BN(20),
+        locked: new BN(20),
+        transferable: new BN(0),
+        recoverable: new BN(0),
+        locks: [],
+      },
```

Comment:
```suggestion
    balances = {
      [alice.address]: mockBalances(10, 0, 10, 0),
      [aliceStash.address]: mockBalances(50, 0, 50, 0),
      [bob.address]: mockBalances(20, 20, 0, 0),
```

WDYT about some helper? The balances object is big and hard for me to check the setup vs the expected part

Reply:
```suggestion
    balances = {
      [alice.address]: {
        ...zeroBalance,
        total: new BN(10),
        transferable: new BN(10),
      },
      [aliceStash.address]: {
        ...zeroBalance,
        total: new BN(50),
        transferable: new BN(50),
      },
      [bob.address]: {
        ...zeroBalance,
        total: new BN(20),
        locked: new BN(20),
      },
```

Or :point\_up: maybe even better? Dunno, :thinking: